src/codeu/chat/client/core/View.java

```diff
@@ -136,4 +139,26 @@ public View(ConnectionSource source) {
 
     return messages;
   }
+  /**
+   * Get the info of the server; version info should be returned currently.
+   * @return The infomation of the server. If fails, null will be returned.
+   */
+  public ServerInfo getInfo() {
+    try (final Connection connection = this.source.connect()) {
+      Serializers.INTEGER.write(connection.out(), NetworkCode.SERVER_INFO_REQUEST);
+      if (Serializers.INTEGER.read(connection.in()) == NetworkCode.SERVER_INFO_RESPONSE) {
+        final Uuid version = Uuid.SERIALIZER.read(connection.in());
+        return new ServerInfo(version);
+      } else {
+        // Communicate this error - the server did not respond with the type of
+        // response we expected.
+        new IOException("ERROR: The server did not respond with the type of response we expected.").printStackTrace();
```

Comment:
That format only works with try/catch(Exception e) scenarios, and can only be implemented in the catch brackets. A simple System.out.println("ERROR: The server did not respond with the type of response we expected."); would do. Also, when I ran this, I got a different error because there are no IOException constructors that just take a String.